### PR TITLE
Tests for demo content

### PR DESCRIPTION
**Pull request by PavelBulat**

### CODE REVIEW

File: tests/unit/falcon_demo_content/DemoContentTest.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+  const DEMO_CONTENT_MODULE = 'falcon_demo_content';
+
+  /**
+   * Checks for the presence demo content.
+   * @group basic
+   */
+  public function testDemoContentEnabled()
+  {
+    // Get array of demo content entities uuid and type.
```
Comment (spleshka) [inline tests/unit/falcon_demo_content/DemoContentTest.php]: Wrong comment.

File: modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      $params['subject'] = $subject;
+      $params['body'] = $body;
+      $params['headers'] = ['Content-Type' => 'text/html'];
+      $params['render_tokens']['commerce_order'] = $order;
+      $params['replace_tokens'] = TRUE;
+
+      $this->pluginManagerMail->mail('falcon_donation', 'donation_thank_you_email', $to, $langcode, $params);
+    }
+    catch (\Exception $e) {
+      $this->logger
```
Comment (spleshka) [inline modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php:107]: Please use proper string replacement. It is insecure to concat strings like this.

File: modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+      $subject = $appeal->get('field_thankyou_email_subject')->value;
+      $body = $appeal->get('field_thankyou_email_body')->value;
+
+      // Check subject and body for thank you email.
+      if (empty($subject) || empty($body)) {
+        return;
+      }
+
+      // Get email.
```
Comment (spleshka) [inline modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php]: Please, don't leave comments like this. Ever.

File: modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        return;
+      }
+
+      $appeal = $order->get('field_appeal')->entity;
+
+      $subject = $appeal->get('field_thankyou_email_subject')->value;
+      $body = $appeal->get('field_thankyou_email_body')->value;
+
+      // Check subject and body for thank you email.
+      if (empty($subject) || empty($body)) {
```
Comment (spleshka) [inline modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php:83]: I don't believe that empty subject should cancel the email sending.

File: modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        return;
```
Comment (spleshka) [inline modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php:89]: If something went wrong it's always a good idea to log it for further debugging.

File: modules/custom/falcon_gift_ecards/src/EventSubscriber/FalconOrderCompletedSubscriber.php [hunk 6]
```diff
@@ -90,7 +99,7 @@ public function orderCompleteHandler(WorkflowTransitionEvent $event) {
       }
     }
     catch (\Exception $e) {
-      \Drupal::logger('falcon_gift_ecards')
+      $this->logger
```
Comment (spleshka) [inline modules/custom/falcon_gift_ecards/src/EventSubscriber/FalconOrderCompletedSubscriber.php:102]: Same issue as before.

File: .circleci/config.yml [hunk 7]
```diff
@@ -50,6 +54,10 @@ jobs:
           name: Run basic and additional groups tests
           command: make tests:run -- -g basic -g additional
 
+      - run:
```
Comment (spleshka) [inline .circleci/config.yml:53]: Not needed step.

File: tests/unit/falcon_demo_content/DemoContentTest.php [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    // Checks for the presence entity in database.
+    foreach ($contentEntities as $contentEntity) {
+      $this->assertNotEmpty($entityRepository->loadEntityByUuid($contentEntity['entity_type_id'], $contentEntity['uuid']),
+        "Missing content with uuid = " . $contentEntity['uuid']);
+    }
+  }
+
+  /**
+   * Checks for the missing demo content.
+   * @group basic
```
Comment (spleshka) [inline tests/unit/falcon_demo_content/DemoContentTest.php]: Please, put it into a separate group `demo_content` and test separately at the end of all other tests.

**Review by spleshka — CHANGES_REQUESTED** [review]

File: tests/unit/falcon_demo_content/DemoContentTest.php (hunk 1 above)
Comment (PavelBulat) [reply tests/unit/falcon_demo_content/DemoContentTest.php]: Deleted.

File: modules/custom/falcon_gift_ecards/src/EventSubscriber/FalconOrderCompletedSubscriber.php (hunk 6 above)
Comment (PavelBulat) [reply modules/custom/falcon_gift_ecards/src/EventSubscriber/FalconOrderCompletedSubscriber.php:102]: Changed.

File: .circleci/config.yml (hunk 7 above)
Comment (PavelBulat) [reply .circleci/config.yml:53]: Removed.

File: tests/unit/falcon_demo_content/DemoContentTest.php (hunk 8 above)
Comment (PavelBulat) [reply tests/unit/falcon_demo_content/DemoContentTest.php]: Done.

File: modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php (hunk 5 above)
Comment (PavelBulat) [reply modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php:89]: Added adding to log.

File: modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php (hunk 4 above)
Comment (PavelBulat) [reply modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php:83]: Discussed.

File: modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php (hunk 3 above)
Comment (PavelBulat) [reply modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php]: Deleted. I will be trying.

File: modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php (hunk 2 above)
Comment (PavelBulat) [reply modules/custom/falcon_donation/src/EventSubscriber/FalconDonationOrderCompletedSubscriber.php:107]: Changed

**Pull request merged**